rosidl_generator_c/resource/full__description.c.em

```diff
@@ -40,7 +41,7 @@ for referenced_td in toplevel_msg['referenced_type_descriptions']:
     names = referenced_td['type_name'].split('/')
     _type = NamespacedType(names[:-1], names[-1])
     include_prefix = idl_structure_type_to_c_include_prefix(_type, 'detail')
-    includes.add(include_prefix + '__functions.h')
+    includes[include_prefix + '__functions.h'] = None
```

Comment:
I believe this line might add the same include twice, the `set` (or `OrderedDict`) is used to deduplicate. So I am not sure a list can be used here.

Reply:
Saying that I tried a (very) basic test and didn't see any duplicate `#includes` so it _might_ be that `toplevel_msg['referenced_type_descriptions']` is already de-deduplicated?

Reply:
Yep I can do this, just want to check we are happy with the performance of `sorted` (copying all the items plus `O(n log n)` sorting).

I originally went with `OrderedDict` because its already used in `*.em` files[1] (although admittedly `OrderedDict` is never used with `None` values) and I was happy that switching from a `set` to an `OrderedDict` would not add any meaningful overhead to compute required for codegen.

---

1: e.g. https://github.com/ros2/rosidl/blob/2f900108d0bf234139c85b25b7531f20b1cb5a51/rosidl_generator_cpp/resource/msg__struct.hpp.em#L35

Reply:
> Yep I can do this, just want to check we are happy with the performance of `sorted` (copying all the items plus `O(n log n)` sorting).

Another good point.

> I originally went with `OrderedDict` because its already used in `*.em` files[1] (although admittedly `OrderedDict` is never used with `None` values) and I was happy that switching from a `set` to an `OrderedDict` would not add any meaningful overhead to compute required for codegen.

The reason that this is causing me to try other things is that an `OrderedDict` with `None` as the values feels like the incorrect structure.  But fair enough, I've tried other things that have other drawbacks, so I'm OK with this.  Thanks for iterating with me, I'm going to withdraw/resolve this comment and run CI here.